### DISCUSSION TITLE
Make the app crash

### DISCUSSION THREAD

**Discussion by massicer** [fastapi/fastapi#2701] [Questions] [answered]
Hello 👋, first of all thank you for the amazing project! It's awesome! 🚀
I'm wondering if there is a native way to make my Fastapi app crash. It could be helpful for example to make possible to kubernetes or other orchestrator to delete and restart the pod where the app is running.

I notice there is a default middleware that replies `500` and avoids the app to crash. There is a way to disable it? Could be it become a future feature to work on?

Just to be clear, here's a simple example:
```
from fastapi import FastAPI
import uvicorn

def build_app():

    app = FastAPI()


    @app.get("/hello")
    def read_root():
        return {"Hello": "World"}


    @app.get("/raise_exception")
    def read_item():
        raise ValueError("My exception")

    return app

if __name__ == "__main__":
    app = build_app()
    uvicorn.run(app, port=8080, log_level="debug")

```

**Comment by Kludex** [comment]
What you want is a "liveness" endpoint to be more precise, right?

This can help: https://pypi.org/project/fastapi-health/ (not necessary, at all, but can help)

**Reply by massicer** [reply]
Thank you for the suggestion , but I was wondering to a way to make the main process of the container crash, in order to be able to collect and track the issues trough an error tracking software like Sentry.

**Comment by kbakk** [comment, marked as answer]
If you want Sentry reporting, you don't have to make the app crash - there's a way to use middleware to report errors:

https://docs.sentry.io/platforms/python/guides/asgi/

**Reply by massicer** [reply]
Thank you! Your suggestion plus the addition of the liveness probe will probably solve my problems.
😊
